src/frontend/components/SubscribeScheduledVideo/index.spec.tsx

```diff
@@ -0,0 +1,429 @@
+              starting_at: startingAt.toISOString(),
+            }}
+          />
+        </Grommet>,
+      ),
+    );
+
+    await screen.findByRole('button', { name: /register/i });
+    screen.getByRole('heading', {
+      name: `Register to this event starting on ${startingAt.toLocaleString()}`,
+    });
+    screen.getByRole('heading', {
+      name: /days left 2 days 0 0 : 0 1 : 0 0/i,
+    });
+    screen.getByRole('textbox', { name: /type your email to subscribe/i });
+    screen.queryByText('Learn about equations');
+    screen.queryByText('Maths');
+    act(() => {
+      // advance time by 2 minutes and one second and few milliseconds to avoid the
+      // div added by the component Clock while digits are changing
+      jest.advanceTimersByTime(1000 * (60 * 2 + 1) + 25);
+    });
+    await screen.findByRole('heading', {
+      name: /days left 1 day 2 3 : 5 8 : 5 9/i,
+    });
+  });
+
+  it('shows singular for 0 day', async () => {
+    mockDecodedJwt = {};
+    const startingAt = new Date();
+    startingAt.setHours(startingAt.getHours() + 11);
+    startingAt.setMinutes(startingAt.getMinutes() + 24);
+    startingAt.setSeconds(startingAt.getSeconds() + 30);
+    render(
+      wrapInIntlProvider(
+        <Grommet>
+          <SubscribeScheduledVideo
+            video={{
+              ...video,
+              starting_at: startingAt.toISOString(),
+            }}
+          />
+        </Grommet>,
+      ),
+    );
+
+    await screen.findByRole('button', { name: /register/i });
+    screen.getByRole('heading', {
+      name: `Register to this event starting on ${startingAt.toLocaleString()}`,
+    });
+    screen.getByRole('heading', {
+      name: /days left 0 day 1 1 : 2 4 : 3 0/i,
+    });
+  });
+
+  it('calls the API if token has email', async () => {
+    mockDecodedJwt = {
+      user: {
+        email: 'chantal@fun-mooc.fr',
+      },
+    };
+    fetchMock.get('/api/liveregistrations/?limit=999', {
+      count: 0,
+    });
+    render(
+      wrapInIntlProvider(
+        <Grommet>
+          <SubscribeScheduledVideo
+            video={{
+              ...video,
+              starting_at: startingAtInTheFuture.toISOString(),
+            }}
+          />
+        </Grommet>,
+      ),
+    );
+    // API gets called
+    await waitFor(() => {
+      expect(
+        fetchMock.called('/api/liveregistrations/?limit=999', {
+          method: 'GET',
+        }),
+      ).toBe(true);
+    });
+  });
+
+  it("doesn't call the API if token is empty", async () => {
+    mockDecodedJwt = {};
+    fetchMock.get('/api/liveregistrations/?limit=999', {
+      count: 0,
+    });
+    render(
+      wrapInIntlProvider(
+        <Grommet>
+          <SubscribeScheduledVideo
+            video={{
+              ...video,
+              starting_at: startingAtInTheFuture.toISOString(),
+            }}
+          />
+        </Grommet>,
+      ),
+    );
+
+    // API doesn't get called
+    await waitFor(() => {
```

Comment:
The way you tested in test `  it("doesn't call the API if token has only context_id")` is I think better. You wait the complete rendering and then you test if the api is not called.